### PR TITLE
Add floppy_files, cd_files, and iso_paths to vsphere-clone

### PR DESCRIPTION
**Pull request by sylviamoss**
This adds the same floppy files, cd files, and iso_paths from vsphere-iso to vsphere-clone, and also a bunch of tests for all the features. 

Closes https://github.com/hashicorp/packer/issues/9868
Closes https://github.com/hashicorp/packer/issues/9979

### CODE REVIEW

**Comment by rconde01** [comment]
By cd files you mean iso_paths?

**Comment by sylviamoss** [comment]
@rconde01 No.  cd_files means this:  https://5f64ee534d96e40007d53af4--packer-www.netlify.app/docs/builders/vmware/vsphere-clone#cdrom-configuration 
Basically another way to add files. Some new OSes don't allow floppy files anymore and this is the new possible way to do it.

File: builder/vsphere/clone/step_add_cdrom.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	return errs
+}
+
+func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	vm := state.Get("vm").(driver.VirtualMachine)
+
+	if s.Config.CdromType == "sata" {
+		if _, err := vm.FindSATAController(); err == driver.ErrNoSataController {
```
Comment (azr) [inline builder/vsphere/clone/step_add_cdrom.go:39]: here, if there's another error than the `driver.ErrNoSataController` should we say something ?

File: builder/vsphere/clone/step_add_cdrom_test.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
+package clone
```
Comment (azr) [inline builder/vsphere/clone/step_add_cdrom_test.go:1]: nice !

**Review by azr — APPROVED** [review]
LGTM !

File: builder/vsphere/clone/step_add_cdrom.go (hunk 1 above)
Comment (sylviamoss) [reply builder/vsphere/clone/step_add_cdrom.go:39]: Not sure why this was implemented this way ignoring any other error, this is how is implemented in vsphere-iso. Maybe is a bug 🤔  but maybe is not and  ErrNoSataController is the only valid error here. I'll look the API code and look for other possible errors.

File: builder/vsphere/clone/builder.go [hunk 3]
```diff
@@ -58,7 +58,7 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 		&common.StepConfigureHardware{
 			Config: &b.config.HardwareConfig,
 		},
-		&StepAddCDRom{
+		&common.StepAddCDRom{
```
Comment (azr) [inline builder/vsphere/clone/builder.go:61]: nice !

**Review by SwampDragons — APPROVED** [review]

**Pull request merged**